setup.sh

```diff
@@ -29,23 +29,24 @@ if test "$llvm_version" != "$expected_version"; then
   exit 1
 fi
 
-if llvm_config --link-static --libs; then
-    default_mode=static
-    support_static_mode=true
-fi
 
 if llvm_config --link-shared --libs; then
-    default_mode=shared
+    default_mode=${default_mode:=shared}
     support_shared_mode=true
 fi
 
+if llvm_config --link-static --libs; then
+    default_mode=${default_mode:=static}
+    support_static_mode=true
+fi
+
 if test -z "$default_mode"; then
     echo "Something is wrong with the llvm-config command provided."
     exit 1
 fi
 
 base_cflags=$(llvm_config --cflags)
-ldflags="$(llvm_config --ldflags) -lstdc++"
+ldflags="$(pkg-config --libs-only-L libzstd) $(llvm_config --ldflags) -lstdc++"
```

Comment:
This seems unrelated, but I needed it for linking to libllvm in static mode to succeed.

Reply:
Is this a LLVM 16 thing? Wouldn't `llvm-config --system-libs` work here instead?

Reply:
`llvm-config --link-static --system-libs` does include zstd, giving `-lm -lz -lzstd -lcurses -lxml2`, so that looks like the way to go when I get back to this.